### PR TITLE
ON-16700 # Added `EmailTemplates` class

### PR DESCRIPTION
**Pull request by kizaonline**

### CODE REVIEW

File: src/classes/EmailTemplates.ts [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+type EmailTemplatesSearchResult = {
+  formsAppEnvironments: EmailTemplateTypes.EmailTemplate[]
+} & BaseSearchResult
+
+type EmailTemplatesSearchOptions = {
+  formsAppEnvironmentId: number
+  limit?: number
+  offset?: number
+}
```
Comment (mymattcarroll) [inline src/classes/EmailTemplates.ts:13]: These should be added to types and exported, not defined here.

File: src/classes/EmailTemplates.ts [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+   * ```javascript
+   * const emailTemplate = await emailTemplates.getEmailTemplate(1)
+   * // Use data here...
+   * ```
+   *
+   * @param id The id of the form submission approval
+   * @returns
+   */
+  async getEmailTemplate(
+    id?: number,
```
Comment (Zaxist) [inline src/classes/EmailTemplates.ts:63]: I think we want to make these required now

File: src/classes/EmailTemplates.ts [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+   *   data,
+   * )
+   * // Use data here...
+   * ```
+   *
+   * @param data The data for the new email template
+   * @returns
+   */
+  async createEmailTemplate(
+    data?: unknown,
```
Comment (Zaxist) [inline src/classes/EmailTemplates.ts:92]: Required, and typed as `NewEmailTemplate`

File: src/classes/EmailTemplates.ts [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+   *   data,
+   * )
+   * // Use data here...
+   * ```
+   *
+   * @param data The data for the new email template
+   * @returns
+   */
+  async updateEmailTemplate(
+    data?: Record<string, unknown>,
```
Comment (Zaxist) [inline src/classes/EmailTemplates.ts:118]: Required and typed

File: src/classes/EmailTemplates.ts [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+   * Example
+   *
+   * ```javascript
+   * await emailTemplates.deleteEmailTemplate(1)
+   * ```
+   *
+   * @param id The id of the form submission approval
+   * @returns
+   */
+  async deleteEmailTemplate(id?: unknown): Promise<void> {
```
Comment (Zaxist) [inline src/classes/EmailTemplates.ts:137]: required, number

**Pull request merged**